Detections/MultipleDataSources/PhishiglinkExecutionObserved.yaml

```diff
@@ -0,0 +1,97 @@
+id: 2fed0668-6d43-4c78-87e6-510f96f12145
+name: Phishing link click observed in Network Traffic
+description: |
+  'The purpose of this content is to identify successful phishing links accessed by users. Once a user clicks on a phishing link, we observe successful network activity originating from non-Microsoft network devices. These devices may include Palo Alto Networks, Fortinet, Check Point, and Zscaler devices.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId:  OfficeATP
+    dataTypes:
+    - SecurityAlert
+  - connectorId:  PaloAltoNetworks
+    dataTypes:
+    - CommonSecurityLog (PaloAlto)
+  - connectorId:  Fortinet
+    dataTypes:
+    - CommonSecurityLog (Fortinet)
+  - connectorId:  CheckPoint
+    dataTypes:
+    - CommonSecurityLog (CheckPoint)
+  - connectorId:  Zscaler
+    dataTypes:
+    - CommonSecurityLog (Zscaler)
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1566
+query: |
+    //Finding MDO Security alerts and extracting the Entities user, Domain, Ip, and URL. 
+    SecurityAlert
+    |where ProviderName =~ "Office 365 Advanced Threat Protection"
+    //extracting Alert Entities 
+    | extend Entities = parse_json(Entities)
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'account'
+    | extend EntityUPN = iff(isempty(Entity.UserPrincipalName), tostring(strcat(Entity.Name, "@", tostring (Entity.UPNSuffix))), tostring(Entity.UserPrincipalName))
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'ip'|extend EntityIp = tostring(Entity.Address)
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'host'| extend EntityHost = tostring(Entity.HostName)
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'url'
+    | extend EntityUrl = tostring(Entity.Url)
+    )
+    | where not(ipv4_is_private(EntityIp))
+    | summarize
+    AccountUpn=take_any(EntityUPN),IPAddress=take_any(EntityIp),HostName=take_any(EntityHost),Url=tostring(tolower(take_any(EntityUrl))),AlertTime= min(TimeGenerated)by SystemAlertId, ProductName
+    // filtering 3pnetwork devices 
+    | join kind= inner (CommonSecurityLog
+    | where DeviceVendor has_any  ("Palo Alto Networks", "Fortinet", "Check Point", "Zscaler")
```

Comment:
Can we perform some time-based filtering here on TimeGenerated before the join occurs?

Reply:
This is enforced at run time with the queryPeriod value